pkg/networking/cilium/installer.go

```diff
@@ -0,0 +1,63 @@
+package cilium
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+// InstallerForSpec allows to configure Cilium for a particular EKS-A cluster
+// It's a stateful version of installer, with a fixed Cilium config
+type InstallerForSpec struct {
+	installer Installer
+	config    Config
+}
+
+func NewInstallerForSpec(client KubernetesClient, templater InstallTemplater, config Config) *InstallerForSpec {
+	return &InstallerForSpec{
+		installer: *NewInstaller(client, templater),
+		config:    config,
+	}
+}
+
+// Installs installs Cilium in an cluster
+func (i *InstallerForSpec) Install(ctx context.Context, cluster *types.Cluster) error {
+	return i.installer.Install(ctx, cluster, i.config.Spec, i.config.AllowedNamespaces)
+}
+
+// InstallTemplater generates a Cilium manifest for installation
+type InstallTemplater interface {
+	GenerateManifest(ctx context.Context, spec *cluster.Spec, opts ...ManifestOpt) ([]byte, error)
+}
+
+// Installer allows to configure Cilium in a cluster
+type Installer struct {
+	templater InstallTemplater
+	k8s       KubernetesClient
+}
+
+func NewInstaller(client KubernetesClient, templater InstallTemplater) *Installer {
+	return &Installer{
+		templater: templater,
+		k8s:       client,
+	}
+}
+
+// Installs configures Cilium in an EKS-A cluster
+func (i *Installer) Install(ctx context.Context, cluster *types.Cluster, spec *cluster.Spec, allowedNamespaces []string) error {
```

Comment:
Is this behavior needed by anything? If not would it make sense to have a single `Installer` type exporting the `Install` behavior expected by `workflow` constructs and an unexported `install` (this behavior) reducing the API surface area?

E.g.
```
type Installerr struct {
	Templater
	allowedNamespaces []string
	spec              *cluster.Spec
	k8s               KubernetesClient
}

func (i Installerr) Install(ctx context.Context, cluster *types.Cluster) error {
	// Similar impl
}
```

Reply:
The current workflow uses this (through cluster manager)